### PR TITLE
Use requestAnimationFrame instead of $timeout

### PR DESCRIPTION
**Pull request by juherpin**
When using $timeout to trigger the animation, some browsers (eg. Chrome for Android) are not showing the entering animation.
With the requestAnimationFrame polyfill it works.

### CODE REVIEW

File: js/ngDialog.js [hunk 1]
```diff
@@ -26,6 +26,12 @@
 	var animationEndSupport = isDef(style.animation) || isDef(style.WebkitAnimation) || isDef(style.MozAnimation) || isDef(style.MsAnimation) || isDef(style.OAnimation);
 	var animationEndEvent = 'animationend webkitAnimationEnd mozAnimationEnd MSAnimationEnd oanimationend';
 	var focusableElementSelector = 'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, *[tabindex], *[contenteditable]';
+	var reqAnimationFrame = window.requestAnimationFrame ||
+		window.webkitRequestAnimationFrame ||
+		window.mozRequestAnimationFrame    ||
+		function( callback ){
```
Comment (voronianski) [inline js/ngDialog.js]: @juherpin please fix code style issues (see [CONTRIBUTING.md](https://github.com/likeastore/ngDialog/blob/master/CONTRIBUTING.md)).

File: js/ngDialog.js [hunk 2]
```diff
@@ -26,6 +26,12 @@
 	var animationEndSupport = isDef(style.animation) || isDef(style.WebkitAnimation) || isDef(style.MozAnimation) || isDef(style.MsAnimation) || isDef(style.OAnimation);
 	var animationEndEvent = 'animationend webkitAnimationEnd mozAnimationEnd MSAnimationEnd oanimationend';
 	var focusableElementSelector = 'a[href], area[href], input:not([disabled]), select:not([disabled]), textarea:not([disabled]), button:not([disabled]), iframe, object, embed, *[tabindex], *[contenteditable]';
+	var reqAnimationFrame = window.requestAnimationFrame ||
```
Comment (voronianski) [inline js/ngDialog.js]: @juherpin we have angular's `$window` injected, I think it's better to use it instead.

File: js/ngDialog.js [hunk 3]
```diff
@@ -26,6 +26,12 @@
...
+		window.webkitRequestAnimationFrame ||
+		window.mozRequestAnimationFrame    ||
```
Comment (voronianski) [inline js/ngDialog.js]: @juherpin where is `oRequestAnimationFrame` and `msRequestAnimationFrame` ?

File: js/ngDialog.js (hunk 2 above)
Comment (juherpin) [inline js/ngDialog.js]: $window is not injected at this line. I can move the variable "reqAnimationFrame" into the $get method but i think it should be better to be declared after animationEndSupport, animationEndEvent, ...

**Comment by voronianski** [comment]
@juherpin please rebase with latest `master` and I'll merge this inside.

**Comment by voronianski** [comment]
@juherpin I've tried your solution however it breaks tests, please rebase with latest master and make sure the build is passing, thanks.

**Pull request closed without merging**